doc/contributing/primordials.md

```diff
@@ -360,13 +360,33 @@ Object.defineProperty(Object.prototype, Symbol.isConcatSpreadable, {
          <code>%Promise.race%</code> iterate over an array</summary>
 
 ```js
+const array = [promise];
+const set = new SafeSet().add(promise);
+
 // 1. Lookup @@iterator property on `array` (user-mutable if user-provided).
 // 2. Lookup @@iterator property on %Array.prototype% (user-mutable).
 // 3. Lookup `next` property on %ArrayIteratorPrototype% (user-mutable).
+// 4. Lookup `then` property on `promise` (user-mutable if user-provided).
+// 5. Lookup `then` property on `%Promise.prototype%` (user-mutable).
 PromiseAll(array); // unsafe
 
-PromiseAll(new SafeArrayIterator(array));
+PromiseAll(new SafeArrayIterator(array)); // unsafe
```

Comment:
question: it is unsafe because of the following steps, right?

4. Lookup `then` property on `promise` (user-mutable if user-provided).
5. Lookup `then` property on `%Promise.prototype%` (user-mutable).

Reply:
Yes that's correct 👍 I thought of updating the comment to point it out, but I felt it was just repeating the same thing and leave it at that, but open to suggestions.

Reply:
I would prefer to point it out😁, it does take me some time to figure out why this usage is unsafe.